api/src/test/java/org/openmrs/test/jupiter/BaseContextSensitiveTest.java

```diff
@@ -0,0 +1,1027 @@
+
+				updateSearchIndex();
+				
+				isBaseSetup = true;
+			}
+			
+			authenticate();
+		} else {
+			if (isBaseSetup) {
+				deleteAllData();
+			}
+		}
+		
+		Context.clearSession();
+	}
+	
+	public Class<?>[] getIndexedTypes() {
+		return new Class<?>[] { ConceptName.class, Drug.class, PersonName.class, PersonAttribute.class,
+				PatientIdentifier.class};
+	}
+	
+	/**
+	 * It needs to be call if you want to do a concept search after you modify a concept in a test.
+	 * It is because index is automatically updated only after transaction is committed, which
+	 * happens only at the end of a test in our transactional tests.
+	 */
+	public void updateSearchIndex() {
+		for (Class<?> indexType : getIndexedTypes()) {
+			Context.updateSearchIndexForType(indexType);
+		}
+	}
+	
+	@AfterEach
+	public void clearSessionAfterEachTest() {
+		// clear the session to make sure nothing is cached, etc
+		Context.clearSession();
+		
+		// needed because the authenticatedUser is the only object that sticks
+		// around after tests and the clearSession call
+		if (Context.isSessionOpen())
+			Context.logout();
+	}
+	
+	/**
+	 * Called after each test class. This is called once per test class that extends
+	 * {@link BaseContextSensitiveTest}. Needed so that "unit of work" that is the test class is
+	 * surrounded by a pair of open/close session calls.
+	 * 
+	 * @throws Exception
+	 */
+	@AfterAll
+	public static void closeSessionAfterEachClass() throws Exception {
+		//Some tests add data via executeDataset()
+		//We need to delete it in order not to interfere with others
+		if (instance != null) {
+			try {
+				instance.deleteAllData();
+			}
+			catch (Exception ex) {
+				//No need to worry about this
+			}
+			instance = null;
+		}
+		
+		// clean up the session so we don't leak memory
+		if (Context.isSessionOpen()) {
+			Context.closeSession();
+		}
+	}
+	
+	/**
+	 * Instance variable used by the {@link #baseSetupWithStandardDataAndAuthentication()} method to
+	 * know whether the current "@Test" method has asked to be _not_ do the initialize/standard
+	 * data/authenticate
+	 * 
+	 * @see SkipBaseSetup
+	 * @see SkipBaseSetupAnnotationExecutionListener
+	 * @see #baseSetupWithStandardDataAndAuthentication()
+	 */
+	private boolean skipBaseSetup = false;
+	
+	/**
+	 * Don't run the {@link #setupDatabaseWithStandardData()} method. This means that the associated
+	 * "@Test" must call one of these:
+	 * 
+	 * <pre>
+	 *  * initializeInMemoryDatabase() ;
+	 *  * executeDataSet(EXAMPLE_DATA_SET);
+	 *  * Authenticate
+	 * </pre>
+	 * 
+	 * on its own if any of those results are needed. This method is called before all "@Test"
+	 * methods that have been annotated with the "@SkipBaseSetup" annotation.
+	 * 
+	 * @throws Exception
+	 * @see SkipBaseSetup
+	 * @see SkipBaseSetupAnnotationExecutionListener
+	 * @see #baseSetupWithStandardDataAndAuthentication()
+	 */
+	public void skipBaseSetup() throws Exception {
+		skipBaseSetup = true;
+	}
+	
+}
```

Comment:
Jupiter support starts from 2.4.x or even 2.5.x. You can remove this file.

Reply:
ok thanks let me collect it

Reply:
Finished thanks

Reply:
I think now its LGTM